src/ShaderGen/FunctionCallGraphDiscoverer.cs

```diff
@@ -123,12 +134,31 @@ public MethodWalker(FunctionCallGraphDiscoverer discoverer) : base(SyntaxWalkerD
                 _discoverer = discoverer;
             }
 
-            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            public override void VisitObjectCreationExpression(ObjectCreationExpressionSyntax node)
             {
-                if (node.ToFullString().Contains("RandomInUnitSphere"))
+                SymbolInfo symbolInfo = _discoverer.Compilation.GetSemanticModel(node.SyntaxTree).GetSymbolInfo(node);
+                ISymbol symbol = symbolInfo.Symbol;
+                if (symbol == null && symbolInfo.CandidateSymbols.Length == 1)
                 {
-
+                    symbol = symbolInfo.CandidateSymbols[0];
                 }
+                if (symbol == null)
+                {
+                    throw new ShaderGenerationException($"A constructor reference could not be identified: {node}");
+                }
+
+                string containingType = symbol.ContainingType.ToDisplayString();
+                _children.Add(new TypeAndMethodName() { TypeName = containingType, MethodName = ".ctor" });
+
+                base.VisitObjectCreationExpression(node);
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                //if (node.ToFullString().Contains("RandomInUnitSphere"))
```

Comment:
nit: remove

Reply:
You added the empty check in 0795e09eb446ccb38a746162ce8964d5c7d60e17 and so I was concerned about just removing it in case it was there to remind you of something?  I'll remove now.

Reply:
Haha, it must have been leftover debug code that I forgot to remove myself. Feel free to remove it 😄